cmd/ingester/app/processor/span_processor.go

```diff
@@ -26,7 +31,30 @@ type SpanProcessor interface {
 	io.Closer
 }
 
+type spanProcessor struct {
+	unmarshaller kafka.Unmarshaller
+	writer       spanstore.Writer
+	io.Closer
+}
+
 // Message contains the fields of the kafka message that the span processor uses
 type Message interface {
 	Value() []byte
 }
+
+// NewSpanProcessor creates a new SpanProcessor
+func NewSpanProcessor(writer spanstore.Writer) SpanProcessor {
+	return &spanProcessor{
+		unmarshaller: kafka.NewProtobufUnmarshaller(),
```

Comment:
I recommend making the unmarshaller configurable by using an `Option`.

That way, we can default to protobuf, but allow people to easily conveniently override it to anything else.